### ISSUE TITLE
Meet NaN loss during finetuning when setting --logging_steps=1

### ISSUE THREAD

**Issue by acphile** [huggingface/optimum-neuron#792]
### System Info

```shell
> apt list | grep neuron | grep installed
aws-neuronx-collectives/unknown,now 2.22.26.0-17a033bc8 amd64 [installed]
aws-neuronx-runtime-lib/unknown,now 2.22.14.0-6e27b8d5b amd64 [installed]
aws-neuronx-tools/unknown,now 2.19.0.0 amd64 [installed]
> python3 -m pip list | grep torch
torch                         2.1.2
torch-neuronx                 2.1.2.2.1.0
torch-xla                     2.1.2
torchvision                   0.16.2
> python3 -m pip list | grep neuron
aws-neuronx-runtime-discovery 2.9
libneuronxla                  2.0.2335
neuronx-cc                    2.14.227.0+2d4f85be
neuronx-distributed           0.8.0
optimum-neuron                0.0.24
torch-neuronx                 2.1.2.2.1.0
transformers-neuronx          0.11.351
```

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction (minimal, reproducible, runnable)

Hi, I am trying to finetune a `mistralai/Mistral-7B-v0.3` based on `optimum.neuron`. When I set the training argument `--logging_steps` to be 1, I will see NaN loss after the second step:
```
{'loss': nan, 'learning_rate': 2.9411764705882356e-07, 'epoch': 0.0}                                                                                                                                                                 
{'loss': nan, 'learning_rate': 4.4117647058823536e-07, 'epoch': 0.0}  
```
However, when I set `--logging_steps=10` while fixing other hyperparameters, the loss is normal. 

The training function is very simple and we just leverage the provided `Trainer` and training arguments for finetuning

```
from optimum.neuron import NeuronHfArgumentParser as HfArgumentParser
from optimum.neuron import NeuronTrainer as Trainer
from optimum.neuron import NeuronTrainingArguments as TrainingArguments

def training_function(
    tokenizer,
    script_args,
    training_args,
    train_dataset,
):

    with lazy_load_for_parallelism(
        tensor_parallel_size=training_args.tensor_parallel_size
    ):
        model = AutoModelForCausalLM.from_pretrained(script_args.model_id, cache_dir="/mnt_out/philepei/instability/modelcache")

    trainer = Trainer(
        model=model,
        tokenizer=tokenizer,
        args=training_args,
        train_dataset=train_dataset,
        # data_collator=data_collator_with_padding,
    )

    # Start training
    trainer.train()
    trainer.save_model()

def main():
    parser = HfArgumentParser([ScriptArguments, TrainingArguments])
    script_args, training_args = parser.parse_args_into_dataclasses()
    ...
    training_function(
        tokenizer,
        script_args,
        training_args,
        tokenized_train_dataset,
    )
```

### Expected behavior

`logging_steps=1` should not lead to NaN loss
